src/track/beats.h

```diff
@@ -56,128 +85,238 @@ class Beats : public QObject {
 
     ////////////////////////////////////////////////////////////////////////////
     // Beat calculations
     ////////////////////////////////////////////////////////////////////////////
 
-    // TODO: We may want all of these find functions to return an integer
-    //       instead of a double.
     // TODO: We may want to implement these with common code that returns
     //       the triple of closest, next, and prev.
 
-    // Starting from sample dSamples, return the sample of the next beat in the
-    // track, or -1 if none exists. If dSamples refers to the location of a
-    // beat, dSamples is returned.
-    virtual double findNextBeat(double dSamples) const = 0;
+    // Starting from frame, return the frame number of the next beat
+    // in the track, or -1 if none exists. If frame refers to the location
+    // of a beat, frame is returned.
+    virtual double findNextBeat(double sample) const {
+        return findNextBeatNew(sample / 2.0) * 2.0;
+    }
+    virtual FrameNum findNextBeatNew(FrameNum frame) const;
 
-    // Starting from sample dSamples, return the sample of the previous beat in
-    // the track, or -1 if none exists. If dSamples refers to the location of
-    // beat, dSamples is returned.
-    virtual double findPrevBeat(double dSamples) const = 0;
+    // Starting from frame frame, return the frame number of the previous
+    // beat in the track, or -1 if none exists. If frame refers to the
+    // location of beat, frame is returned.
+    virtual double findPrevBeat(FrameNum frame) const {
+        return findPrevBeatNew(frame / 2.0) * 2.0;
+    }
+    virtual FrameNum findPrevBeatNew(FrameNum frame) const;
 
-    // Starting from sample dSamples, fill the samples of the previous beat
-    // and next beat.  Either can be -1 if none exists.  If dSamples refers
-    // to the location of the beat, the first value is dSamples, and the second
+    // Starting from frame, fill the frame numbers of the previous beat
+    // and next beat.  Either can be -1 if none exists.  If frame refers
+    // to the location of the beat, the first value is frame, and the second
     // value is the next beat position.  Non- -1 values are guaranteed to be
     // even.  Returns false if *at least one* sample is -1.  (Can return false
     // with one beat successfully filled)
-    virtual bool findPrevNextBeats(double dSamples,
-                                   double* dpPrevBeatSamples,
-                                   double* dpNextBeatSamples) const = 0;
+    virtual bool findPrevNextBeats(double sample,
+            double* pPrevSample,
+            double* pNextSample) const {
+        FrameNum prev, next;
+        bool result;
+
+        result = findPrevNextBeatsNew(sample / 2.0, &prev, &next);
+        *pPrevSample = prev * 2.0;
+        *pNextSample = next * 2.0;
+        return result;
+    }
+    virtual bool findPrevNextBeatsNew(FrameNum frame,
```

Comment:
I suggest renaming the legacy functions like `findPrevNextBeatsSample` and removing "New" from from the names of the new ones.

Reply:
Actually, don't do that. It will create merge conflicts with other pending PRs like #2194.

Reply:
Let's use "Frame" instead of "New" for the new functions.

Reply:
Hummm. My idea was to review all the code that uses any Beats method and adapt them to Frames. Once that is done I can get rid of the adaptation methods and remove the New from the methods. I feel that I don't do the adaptation of all that code we will end with a dual intarface samples/frames for ever.
As the assumption is always stereo, now is easy to modify the functions in the rest of the code.
What do you think?

Reply:
I agree with moving the the new frame API ASAP. However I already know this will create lots of tedious merge conflicts with #2194, so skip moving CueControl and LoopingControl to the new API.

Reply:
Nevermind, I took a closer look at the changes in #2194 and they are not in the parts of the code that interact with the Beats class. So go ahead and move as much as you can to the new frame API.

Reply:
I have renamed the new functions with Frames suffix.
I will remove the adaptation functions later on.